src/Workspaces/MSBuildTest/OpenProjectsTests.cs

```diff
@@ -0,0 +1,225 @@
+        }
+
+        public static TheoryData<string> GetCSharpProjectTemplateNames()
+            => GetProjectTemplateNames("c#");
+
+        public static TheoryData<string> GetVisualBasicProjectTemplateNames()
+            => GetProjectTemplateNames("vb");
+
+        public static TheoryData<string> GetProjectTemplateNames(string language)
+        {
+            // The expected output from the list command is as follows.
+
+            // These templates matched your input: --language='vb', --type='project'
+            //
+            // Template Name                  Short Name           Language  Tags
+            // -----------------------------  -------------------  --------  ---------------
+            // Class Library                  classlib             C#,F#,VB  Common/Library
+            // Console App                    console              C#,F#,VB  Common/Console
+            // ...
+
+            var result = RunDotNet($"new list --type project --language {language}");
+            Assert.Equal(0, result.ExitCode);
+
+            var lines = result.Output
+                .Replace("\r\n", "\n")
+                .Split(new[] { "\n" }, StringSplitOptions.RemoveEmptyEntries);
+
+            TheoryData<string> templateNames = [];
+            var foundDivider = false;
+
+            foreach (var line in lines)
+            {
+                if (!foundDivider)
+                {
+                    if (line.StartsWith("----"))
+                    {
+                        foundDivider = true;
+                    }
+                    continue;
+                }
+
+                var columns = line.Split(new[] { "  " }, StringSplitOptions.RemoveEmptyEntries)
+                    .Select(c => c.Trim())
+                    .ToArray();
+                var templateShortName = columns[1].Split(',').First();
+
+                if (ExcludeMauiTemplates && templateShortName.StartsWith("maui"))
+                    continue;
+
+                templateNames.Add(templateShortName);
+            }
+
+            return templateNames;
+        }
+
+        private async Task AssertTemplateProjectLoadsCleanlyAsync(string templateName, string languageName, string[]? ignoredDiagnostics = null)
+        {
+            if (ignoredDiagnostics?.Length > 0)
+            {
+                TestOutputHelper.WriteLine($"Ignoring compiler diagnostics: \"{string.Join("\", \"", ignoredDiagnostics)}\"");
+            }
+
+            using var projectDirectory = (DisposableDirectory)SolutionDirectory;
+
+            var projectFilePath = GenerateProjectFromTemplate(projectDirectory.Path, templateName, languageName, TestOutputHelper);
+
+            await AssertProjectLoadsCleanlyAsync(projectFilePath, ignoredDiagnostics ?? []);
+
+            return;
+
+            static string GenerateProjectFromTemplate(string projectPath, string templateName, string languageName, ITestOutputHelper outputHelper)
```

Comment:
Consider just inlining this -- once you add up the method call and the return, there's a lot more boilerplate than just inlining the call to CreateNewProject.